src/Lecture1.hs

```diff
@@ -108,7 +126,12 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+
+strSum :: String -> Int
+strSum str = sum(stringToInt(words str))
+    where
+        stringToInt :: [String] -> [Int]
+        stringToInt = map read
```

Comment:
As you specify the top-level type signature for this function, the compiler would be able to figure out what type `read` function has in this case due to type inference, so you can skip specifying it in the body or by writing a helper function.

After that, you're able to implement `strSum` as a neat one-liner 🙂 


```suggestion
strSum str = sum (map read (words str))
```